src/utils/hooks/use-redirect-location.js

```diff
@@ -144,3 +155,123 @@ export const useRedirectLocationInFlowchart = (
+  });
+
+  const matchedSelectedView = matchPath(pathname + search, {
+    exact: true,
+    path: [routes.experimentTracking.selectedView],
+  });
+
+  const matchedSelectedRuns = matchPath(pathname + search, {
+    exact: true,
+    path: [routes.experimentTracking.selectedRuns],
+  });
+
+  useEffect(() => {
+    if (
+      !matchedExperimentTrackingMainPage &&
+      !matchedSelectedRuns &&
+      !matchedSelectedView
+    ) {
+      setErrorMessage(errorMessages.experimentTracking);
+      setInvalidUrl(true);
+    }
+
+    if (matchedExperimentTrackingMainPage) {
+      if (data?.runsList.length > 0 && selectedRunIds.length === 0) {
+        setErrorMessage({});
+        setInvalidUrl(false);
+
+        /**
+         * If we return runs and don't yet have a selected run, set the first one
+         * as the default, with precedence given to runs that are bookmarked.
+         */
+        const bookmarkedRuns = data.runsList.filter((run) => {
+          return run.bookmark === true;
+        });
+
+        if (bookmarkedRuns.length > 0) {
+          const defaultRunFromBookmarked = bookmarkedRuns
+            .map((run) => run.id)
+            .slice(0, 1);
+          setSelectedRunIds(defaultRunFromBookmarked);
+          toSelectedRunsPath(
+            defaultRunFromBookmarked,
+            activeTab,
+            enableComparisonView
+          );
+        } else {
+          const defaultRun = data.runsList.map((run) => run.id).slice(0, 1);
+          setSelectedRunIds(defaultRun);
+          toSelectedRunsPath(defaultRun, activeTab, enableComparisonView);
+        }
+      }
+    }
+
+    if (matchedSelectedRuns && data) {
+      const { params: searchParams } = matchedSelectedRuns;
+
+      const runIdsArray = searchParams.ids.split(',');
+
+      const allRunIds = data?.runsList.map((run) => run.id);
+      const notFoundIds = runIdsArray.find((id) => !allRunIds?.includes(id));
+
+      // Extra check if the ids from URL are not existed
+      if (notFoundIds) {
+        setErrorMessage(errorMessages.runIds);
+        setInvalidUrl(true);
+      } else {
+        const view = getDefaultTabLabel(searchParams);
+
+        // If there is more than 1 runId, the comparison mode should always be true
+        const isComparison =
+          runIdsArray.length > 1 ? 'true' : searchParams.isComparison;
+
+        setSelectedRunIds(runIdsArray);
+        setEnableComparisonView(isComparison === 'true');
+        setActiveTab(view);
+      }
+    }
+
+    /**
+     * This is for when there's only view= is defined in the URL, without any run_ids
+     * it should re-direct to the latest run
+     */
+    if (matchedSelectedView && data) {
+      const { params } = matchedSelectedView;
+
+      const latestRun = data.runsList.map((run) => run.id).slice(0, 1);
+      setSelectedRunIds(latestRun);
+
+      const view = getDefaultTabLabel(params);
+      toSelectedRunsPath(latestRun, view, enableComparisonView);
```

Comment:
Group expressions and function calls together.

```suggestion
      const { params } = matchedSelectedView;
      const latestRun = data.runsList.map((run) => run.id).slice(0, 1);
      const view = getDefaultTabLabel(params);
      
      setSelectedRunIds(latestRun);
      toSelectedRunsPath(latestRun, view, enableComparisonView);
```